manifests/server/config.pp

```diff
@@ -256,12 +258,15 @@
           content => template('postgresql/systemd-override.erb'),
           require => File['systemd-conf-dir'],
       ;
+    }
 
-      # Remove old unit file to avoid conflicts
-      'old-systemd-override':
-          ensure => absent,
-          path   => "/etc/systemd/system/${service_name}.service",
-      ;
+    # Remove old unit file to avoid conflicts
+    file {'old-systemd-override':
+      ensure => absent,
+      owner  => root,
+      group  => root,
```

Comment:
Specifying this on an `ensure => absent` isn't needed.

Reply:
Thanks @ekohl , yeah from puppet 6 daemon-reloaded is not required. 
The error was seen on upgrade from 6.8.0 to latest version 6.9.0  for puppet agent 6.20 where the daemon-reload was not executed before starting the service. I guess thats due to ordering was not specified which is not fixed.

Above code is modified and added else condition for puppet version > 6 .1.0

It will be used across all file resources systemd-conf-dir/systemd-override/old-systemd-override